utils/src/modules/meter.rs

```diff
@@ -397,33 +399,220 @@ unsafe fn shield_module_send_shield_attack_collision_event(shield_module: *mut u
+            self.x_launch_speed = self.x_launch_speed.abs() - self.decay_x;
+            if self.x_launch_speed < 0.0 {
+                self.x_launch_speed = 0.0;
+            } else {
+                self.x_launch_speed *= dir;
+            }
+        }
+        if self.y_launch_speed != 0.0 {
+            let dir = self.y_launch_speed.signum();
+            self.y_launch_speed = self.y_launch_speed.abs() - self.decay_y;
+            if self.y_launch_speed < 0.0 {
+                self.y_launch_speed = 0.0;
+            } else {
+                self.y_launch_speed *= dir;
+            }
+        }
+        if self.is_damage_fly_top {
+            self.y_chara_speed -= self.damageflytop_gravity;
+            if self.y_chara_speed < -self.damageflytop_fall_speed {
+                self.y_chara_speed = -self.damageflytop_fall_speed;
+            }
+        } else {
+            self.y_chara_speed -= self.gravity;
+            if self.y_chara_speed < -self.fall_speed {
+                self.y_chara_speed = -self.fall_speed;
+            }
+        }
+    }
+}
 
-// extern "C" {
-//     #[link_name = "calculate_finishing_hit"]
-//     fn calculate_finishing_hit(defender: u32, attacker: u32, knockback_info: u64);
-// }
+#[repr(simd)]
+#[derive(Debug)]
+struct Rect {
+    left: f32,
+    right: f32,
+    top: f32,
+    bottom: f32,
+}
 
-// #[skyline::hook(offset = 0x403930, inline)]
-// unsafe fn process_knockback(ctx: &InlineCtx) {
-//     if let Some((defender, attacker)) = IS_CALCULATING {
-//         let boma = *ctx.registers[20].x.as_ref() as *mut smash::app::BattleObjectModuleAccessor;
-//         if (*boma).battle_object_id == defender {
-//             calculate_finishing_hit(defender, attacker, *ctx.registers[19].x.as_ref());
-//         }
-//     }
-// }
+impl Rect {
+    fn contains(&self, x: f32, y: f32) -> bool {
+        (self.left <= x && x <= self.right) && (self.bottom <= y && y <= self.top)
+    }
+}
+
+extern "C" {
+    #[link_name = "_ZN3app6camera13get_dead_areaEv"]
+    fn get_dead_area() -> Rect;
+
+    #[link_name = "_ZN3app10sv_animcmd25EFFECT_GLOBAL_BACK_GROUNDEP9lua_State"]
+    fn EFFECT_GLOBAL_BACK_GROUND(lua_state: u64);
+}
+
+/// Knockback log
+/// 0x8a -> the opponent was grounded (bool)
+/// 0x90 -> backslash (bool)
+/// 0x60 -> stop delay (f32) 
+/// 0x50 -> collision attr (Hash40)
+/// 0x40 -> launch angle in rad (f32)
+/// 0x4 -> level (?)
+/// 0x4c -> hitstop frame
+
+const HANDLE: i32 = 0x01FF;
+const COUNTER: i32 = 0x01FE;
+
+pub unsafe extern "C" fn calculate_finishing_hit(defender: u32, attacker: u32, knockback_info: *const f32) {
+    *(knockback_info.add(0x4c / 4) as *mut u32) = 60;
+    let defender_boma = &mut *(*utils_dyn::util::get_battle_object_from_id(defender)).module_accessor;
+    let attacker_boma = &mut *(*utils_dyn::util::get_battle_object_from_id(attacker)).module_accessor;
+    if !defender_boma.is_fighter() && !defender_boma.is_weapon() { return; }
```

Comment:
Should be `if !defender_boma.is_fighter()` only? Do you want this to run for weapons?